### DISCUSSION TITLE
Going faster in barotrauma is actually going slower?

### DISCUSSION THREAD

**Discussion by duarm** [FakeFishGames/Barotrauma#11618] [Ideas and suggestions]
This is part of the "Mission pacing" problem I've raised in another discussion.

## The problem
What  I mean by the title is, If you try to speed up the mission by going faster with your sub, You'll end up luring every single monster you come across, due to unbalanced spawns and the nature of some monsters, you'll end up reaching your destination much slower than going painfully slow to each objetive and praying to not bump into a dangerous monsters which are not worth killing.

## Possible solutions
There are multiple ways of solving this, which can be cherrypicked, or better, all of them can be tweaked and implemented at the same time.

- **More monsters should announce their presence**, currently, only abyssal monsters announce their presence, an interesting exception are molochs, you often find their babies first, so you know they're around. There should be more of this, especially later on, where each encounter is will most likely sink you because of the sheer number of monsters (hammerheads). Maybe if there are 4+ hammerheads, there's blood nearby, so you know they're ahead, the captain can't see the blood, so someone would need to be on the periscope. 
- **Decrease the noise of the engine by A LOT**, or monster hearing
- **Change map generation to be a diagonal descent**, so the captain can control how much noise the sub makes by balancing sideways/downward movement, it's also thematically fitting.
- **Give the choice to the player: Have different engines which can be bought:** Slower 30% max speed but completely silent engine (Due to the reasons above, this engine would be objectively better than every other, so it needs more tweaking), a middle-ground engine.
- **Decrease spawn rates (of bigger monsters) and increase obstacles**
-  **Have an interesting way to avoid combat for each monster**, e.g. make hammerheads deaf, so if you turn your sonar, you go away. Decrease Molochs hearing, so if you turn both your sonar and go slow, you can avoid it, and so on.
- **Add a depth charge to every sub + other ways of avoiding fight**,  all subs should have access to the interesting depth charge mechanic, no compensation can justify not having the option to use a depth charge when appropriate.

**Comment by Rotiart-Drol** [comment]
I must disagree on some of these points. Stealth is absolutely an option, however it is good that it is reserved for the most straneus of circumstance. I don't think very many crewmen would have fun if the captain stealthed by every possible fight.

> Decrease the noise of the engine by A LOT, or monster hearing

With some application of skill, you can circumvent this. Rely more on vertical movement when possible, as it's completely silent. If you must use the engine, use it gently.

> Add a depth charge to every sub + other ways of avoiding fight, all subs should have access to the interesting depth charge mechanic, no compensation can justify not having the option to use a depth charge when appropriate.

Part of what makes the vanilla subs balanced is their give-and-take approach to equipment. If a submarine doesn't have a depth decoy, it probably has something else worthwhile. Furthermore any submarine with a railgun can be turned into a make shift decoy launcher. Giving every sub a depth decoy would in my opinion take more intrigue out of the game than it would add

**Comment by duarm** [comment]
> With some application of skill, you can circumvent this. Rely more on vertical movement when possible, as it's completely silent. If you must use the engine, use it gently.

I could agree with this if the maps were paced, but given the randomized nature of maps, this is rarely possible. It's very rare for maps to setup an interesting section where you can skillfully avoid combat with vertical movement, as most maps require mostly horizontal movement, especially up until the great sea, hence the second suggestion I made on the thread.

> Part of what makes the vanilla subs balanced is their give-and-take approach to equipment. If a submarine doesn't have a depth decoy, it probably has something else worthwhile.

you don't have much choice because of crew size, but the attack subs are clearly the best for obvious reasons.
What i'm arguing is that the depth charge has a very niche use case (basically a decoy launcher), it's kinda useless on it's current state, so it can be an extra global mechanic of the game to spice things up, and add another more game changing equipment to make up for it, exclusive to some subs. Maybe my third suggestion, with some engines exclusive to some subs?

>Furthermore any submarine with a railgun can be turned into a make shift decoy launcher.

didn't know that, by launching a handheld sonar?

**Reply by Rotiart-Drol** [reply]
Yessir, either a sonar or a beacon. Just need to remember to power it on before you launch it. If you use a beacon you can keep track of where's its going on the nav. Very nice little trick, I remember way back in the day when the Doomworm was relatively new I used that strategy to slip by it

**Comment by Ek-Videogames** [comment]
Stealth is a fallacy because monsters can always "Hear" the inert walls of the submarine, so they "hear" the sub even if the sonar and engine are completely off. And - as stated above - the map generated pathway is sufficiently narrow most of the time that you cannot physically "go around" enemies to avoid them unless you've found some way to allow your submarine to phase through solid ice that is... 😆 

Decoys have some issues, it seems that once the projectile is no longer "active" the attraction property stops.. Also once a monster like a hammerhead attacks the decoy 2 or 3 times it seems to lose interest and "forget" that the decoy is making sond, which usually - due to the decoy being close to the sub that fired it - results in the monster in question noticing the submarine above.

**Reply by Regalis11** [reply]
> Stealth is a fallacy because monsters can always "Hear" the inert walls of the submarine, so they "hear" the sub even if the sonar and engine are completely off.

This is not actually true, the monsters can't "hear the walls". If a monster spots the submarine despite all the loud devices being off, it most likely just saw the submarine.

As a side note, it's possible to visualize the sound/visibility ranges of the entities in the submarine with the commands `debugdraw` and `toggleaitargets`. The orange circles indicate sounds, and blue circles indicate how far things can be seen from.

**Reply by ITPhantom-mod** [reply]
this is a old post, but i have to ask. if blue means that's how far things can be seen from, and orange is what can be heard, whats the red circle next to the orange?

**Reply by SomeRandomNoobKekeke** [reply]
According to [this](https://github.com/FakeFishGames/Barotrauma/blob/0aee53976202c7ea5c0bb3f26669a97c8dc2400c/Barotrauma/BarotraumaClient/ClientSource/Characters/AI/AITarget.cs#L30), yellow orange and red circles are all sounds but from different types of entities

**Comment by ghost** [comment]
It's a bit more complicated that "go faster to go slower", I'm afraid.

Scenario 1: you're just trying to go fast.
All power to engines AND PUMPS, constant tinkering with engines AND PUMPS + captain with maxed skills piloting manually, etc.
Why Would Anyone Do That: personnel transport missions - and that's it? Cargo haulers are slow, fastest submarines are not too well-armed to tackle combat missions, digging/researching invalidates whole point of going fast.

Problem 1: landscape. Unless RNG is exceptionally graceful, you'll have to change directions OFTEN. Which negates all speed cheese and puts extra strain on wirings (and, by extent, on fuel consumption).
Problem 2: monster spawns you mentioned. Virtually anything bigger than a crawler can keep up with a sub plowing through undersea's maze of tunnels and spires. Given abundnance of tight spaces, it's also virtually impossible to just rush through everything.
BUT
Most offending landscape pieces are destructible and can be destroyed easily with few shots.
Most monsters don't sink your sub in one hit so if you've reached a checkpoint (level transition), these just despawn and you're safe. Repair, repeat. You HAVE to be armed (and put arms to good use) in hostile waters of Europa, even weakest subs are armed just enough to fend off small game for at least some time..

Looks balanced to me.

Scenario 2: slightly slower, mowing down whatever monsters are in sight.
Less dedication to engines&pumps, more pew-pew.

Same problems, different solution to monster related problem: gun it.
Personnel transport (and overall transportation) becomes way less viable - but all combat missions are your oysters in this case.
You're basically exchanging 24/7 maintenance of engines, pumps, reactors and wirings to 24/7 maintenance of guns and occasional looting of bodies.

Looks balanced to me.

Scenario 3: stealth. Aviod whatever you can, rush whenever you can.

Since for most hostile entities the sub with sonar off is just a weird floating metal chunk, it allows biding some time on alien research, digging and whatever other exploration missions exist. And/or CAREFULLY tiptoeing around using autopilot AND DIRECTIONAL SONAR.
Since you're not engaging 24/7 with whatever you've spotted, you can afford outlasting hostilities with well-managed repairs on top of weaker guns.
Or utilizing weird strategies like actually stunning stuff with Medic's beefed syringe shots and drifting away.
Since haulers tend to have great engine power, the only source of slowdown for you here is inertia. Filling that Berillia to the brim with cargo then carefully adapting on the go is viable, you just won't be able to tackle biggest threats and sweetest combat missions.
But you won't have to do 24/7 maintenance of guns AND wirings either.

Looks balanced to me.

The only thing I 100% agree with in original post is:

> Add a depth charge to every sub + other ways of avoiding fight, all subs should have access to the interesting depth charge mechanic, no compensation can justify not having the option to use a depth charge when appropriate.

With one teenie-weenie proposal from my side:

Convert Depth Charges To **Torpedoes**, Allow Mounting In Front And Back By Default.
(simply **making decoy a warhead for railguns and depth charges**, getting rid of redundanices in depth charges crafting AND adding some extra utility to railguns, can alleviate this particular problem AND provide some interesting gameplay too)
But it's a theme for another, independant, discussion #12332 .

**Comment by duarm** [comment]
added another suggestion which came to mind

- **More monsters should announce their presence**, currently, only abyssal monsters announce their presence, an interesting exception are molochs, you often find their babies first, so you know they're around, there should be more of this. Especially later on, where each encounter will most likely sink you because of the sheer number of monsters (hammerheads). Maybe if there are 4+ hammerheads, there's blood nearby, so you know they're ahead, the captain can't see the blood, so someone would need to be on the periscope.

**Comment by duarm** [comment]
Sometimes I come back to re-read old discussions, and I usually find typos or I rewrite some sentences to sound better, this results in the discussion being bumped to the top, that's not my intention...
